src/composables/transactionList.ts

```diff
@@ -0,0 +1,324 @@
+    limit: number,
+    recent: boolean,
+    address: Encoded.AccountAddress,
+  ): Promise<ITransaction[]> {
+    const transactionState = getOrInitTransactionStateForAccount(address);
+
+    if (transactionState.nextPageUrl === null && !recent) {
+      return [];
+    }
+    const { getMiddleware } = useMiddleware({ store });
+
+    const [
+      normal,
+      pending,
+      tip,
+    ] = await Promise.all([
+      fetchTransactionsFromMiddleware(transactionState, address, recent, limit),
+      fetchPendingTransaction(address),
+      fetchTipWithdrawnTransactions(address, recent),
+    ]);
+
+    const lastPendingTransaction = pending?.[pending.length - 1];
+    if (lastPendingTransaction?.type === AEX9_TRANSFER_EVENT) {
+      const middleware = await getMiddleware();
+      pending[pending.length - 1] = await middleware.getTx(lastPendingTransaction.payload.txHash);
+    }
+
+    let preparedTransactions = [
+      ...pending,
+      ...normal,
+    ]
+      .filter(({ type }) => !type?.startsWith('Internal'))
+      .map((transaction) => ({
+        ...(transaction.payload || transaction),
+        transactionOwner: address,
+        ...(transaction.type === AEX9_TRANSFER_EVENT
+          ? {
+            tx: {
+              ...transaction.payload,
+              callerId: transaction.payload.senderId,
+              type: 'ContractCallTx',
+            },
+            hash: transaction.payload.txHash,
+            incomplete: true,
+          } as ITransaction
+          : {}),
+      }));
+
+    preparedTransactions = uniqBy(preparedTransactions.reverse(), 'hash').reverse();
+    const minMicroTime = Math.min.apply(null, preparedTransactions.map((tx) => tx.microTime));
+    tip.forEach((f) => {
+      if (f.microTime
+          && (
+            minMicroTime < f.microTime
+              || (preparedTransactions.length === 0 && minMicroTime > f.microTime))
+      ) {
+        preparedTransactions.push({ ...f, transactionOwner: address });
+      }
+    });
+    preparedTransactions = orderBy(preparedTransactions, ['microTime'], ['desc']);
+
+    const oldPendingTransactionForAccount: ITransaction[] = (
+      transactions.value[address]?.pending[nodeNetworkId.value!] || []
+    );
+
+    oldPendingTransactionForAccount.forEach(({ hash }) => {
+      if (preparedTransactions.some((tx) => tx.hash === hash && !tx.pending)) {
+        removePendingTransactionByAccount(address, hash);
+      }
+    });
+
+    preparedTransactions = recent
+      ? preparedTransactions.slice(0, limit)
+      : preparedTransactions;
+
+    transactions.value[address].loaded = uniqBy(
+      [...transactions.value[address].loaded, ...preparedTransactions],
+      'hash',
+    );
+
+    return preparedTransactions;
+  }
+
+  watch(transactions, (value) => {
+    setLocalStorageItem(['transactionsState'], value);
+  }, {
+    deep: true, immediate: true,
+  });
+
+  watch(nodeNetworkId, () => {
+    transactions.value = {};
+  });
```

Comment:
What about situations where this composable is not used by any component and user changed the network?

Reply:
i tested it, and i don't found situation where this is not working properly

Reply:
this place was changed, i close that comment